lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -536,17 +537,29 @@ def exec_module(self, **kwargs):
                         self.fail('Unsupported framework {0} for Linux web app.'.format(self.frameworks[0]['name']))
 
                     self.site_config['linux_fx_version'] = (self.frameworks[0]['name'] + '|' + self.frameworks[0]['version']).upper()
+
+                    if self.frameworks[0]['name'] == 'java':
+                        if self.frameworks[0]['version'] != '8':
+                            self.fail("Linux web app only supports java 8.")
+                        if self.frameworks[0]['settings'] and self.frameworks[0]['settings']['java_container'].lower() != 'tomcat':
+                            self.fail("Linux web app only supports tomcat container.")
+
+                        if self.frameworks[0]['settings'] and self.frameworks[0]['settings']['java_container'].lower() == 'tomcat':
```

Comment:
I think second part of if statement will be always true?

Reply:
no. self.frameworks[0]['settings'] could be None. user could specify framework without settings. eg.
```
frameworks:
 - name: node
    version: 6.9
```

Reply:
you mean tomcat? yes

Reply:
I mean there's:

if is_linux:
else:

or I mismatched levels?

Reply:
fixed.

Reply:
yes, because if it's not 'tomcat' it will fail in previous statement

Reply:
want to explicit call out here since the setting of linux_fx_version for linux webapp is  little tricky.